src/components/Search/Input/index.js

```diff
@@ -18,10 +18,13 @@ const Input = ({ device }) => {
 
   useEffect(() => {
     let results = [];
+    const hasAccentuation = new RegExp(
+      /^\w+$/
+    );
 
-    if (searchTerm !== '') {
+    if (searchTerm !== "") {
       results = catalog.filter((product) =>
-        product.name.toLowerCase().includes(searchTerm.toLowerCase())
+        hasAccentuation.test(searchTerm) ? product.name.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "").includes(searchTerm.toLowerCase()) : product.name.toLowerCase().includes(searchTerm.toLowerCase())
```

Comment:
Bru, não funciona se deixarmos apenas a tratativa do `hasAccentuation` como default?

```suggestion
        product.name.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "").includes(searchTerm.toLowerCase())
```

Reply:
Então, acho que é justamente essa a pegadinha @IsabellaSoares , se normalizarmos por default somente a entrada do input, ao invés da base, quando o usuário digitar um termo com acento, essa string  que foi normalizada por default não retornaria resultado uma vez que o termo correspondente na base tem acentuação. Agora se, por outro lado, validarmos se existe acento no termo buscado e aí sim, normalizar product.name na base conseguimos a busca nas duas variações.

Você chegou a testar dessa outra forma?

Reply:
Acabei de testar e funciona sim, com as duas referências normalizadas, tanto termo de busca como product.name na base. Então ficaria:

```js
  useEffect(() => {
    let results = [];

    if (searchTerm !== "") {
      results = catalog.filter((product) =>
        product.name.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "").includes(searchTerm.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, ""))
      );
    }
```

E não precisaríamos validar `hasAccentuation`, só fiquei em dúvida se em termos de performance impacta fazer a transformação de toda a base ao invés da validação somente do termo no evento/estado.

Reply:
@brubriele 
Bru, eu acho que em termos de performance daria na mesma, porque na versão atual fazemos um `if` para toda a base. A menos que a normalização seja mais pesada do que o condicional... Agora fiquei na dúvida rsrs

Reply:
Bom, por mim está ótimo assim! Foi só uma dúvida, mesmo.